src/Scenes/Scene.cpp

```diff
@@ -1186,7 +1274,87 @@ void Scene::toggleHighlight()
     {
         i->setHighlighted(!i->highlighted());
     }
+
+    //
+    // Highlight all zones
+    //
+    // _zone = ...
+
+}
+
+//
+// Determine if move is valid.
+//
+bool Scene::isValidMove(    std::vector<Zone> blockedZones,
+                            float deltaX,
+                            float deltaY    ) 
+{
+
+    //
+    // Get existing current player tile position...
+    //
+
+    sf::Vector2f pTileCoords = getPlayerTile();
+
+    int ox = pTileCoords.x;
+    int oy = pTileCoords.y;
+
+    _logger->debug(
+        "isValidMove() Old x,y: {},{}", 
+            ox, 
+            oy );
+
+    //
+    // Get potential new tile position...
+    //
+    sf::Vector2f pNewTileCoords = getPlayerTile(    deltaX,
+                                                    deltaY  );
+
+    int nx = pNewTileCoords.x;
+    int ny = pNewTileCoords.y;
+
+    _logger->debug(
+        "isValidMove() New x,y: {},{}", 
+            nx,
+            ny );
+
+    for(const auto& zone: blockedZones) {
```

Comment:
When I was thinking about this my plan was to put the x,y cords of the blocked zones tiles into a two dimensional map-like structure so that way the search for a blocked tile could be constant.

For example: you could have something like: `std::map<int, std::set<int>> _bt // blocked tiles` and then you would populate it with the x,y coords of the blocked tiles (this would require some computation but would only have to be done at load). 

So now when the user takes a step, you get the tile the user wants to move to and then see if that tile is in that structure:

```
if (bt.find(x) != bt.end() && bt[x].find(y) != bt[x].end()) // the cord was found, which means its a blocked tile and the user can't step there
```

This should be (log N) as opposed to the code you have which is O(N).

Also, you might want to look around for a better structure than the map of sets I used here, there's probably something better suited to this than this hack I threw together.